sw/device/lib/dif/dif_alert_handler.c

```diff
@@ -478,6 +478,79 @@ dif_result_t dif_alert_handler_configure(
   return kDifOk;
 }
 
+dif_result_t dif_alert_handler_lock_alert(
+    const dif_alert_handler_t *alert_handler, dif_alert_handler_alert_t alert) {
+  if (alert_handler == NULL || alert >= ALERT_HANDLER_PARAM_N_ALERTS) {
+    return kDifBadArg;
+  }
+
+  ptrdiff_t regwen_offset =
+      ALERT_HANDLER_ALERT_REGWEN_0_REG_OFFSET + alert * sizeof(uint32_t);
+  mmio_region_write32(alert_handler->base_addr, regwen_offset, 0);
+
+  return kDifOk;
+}
+
+dif_result_t dif_alert_handler_is_alert_locked(
+    const dif_alert_handler_t *alert_handler, dif_alert_handler_alert_t alert,
+    bool *is_locked) {
+  if (alert_handler == NULL || alert >= ALERT_HANDLER_PARAM_N_ALERTS ||
+      is_locked == NULL) {
+    return kDifBadArg;
+  }
+
+  ptrdiff_t regwen_offset =
+      ALERT_HANDLER_ALERT_REGWEN_0_REG_OFFSET + alert * sizeof(uint32_t);
+  *is_locked = !mmio_region_read32(alert_handler->base_addr, regwen_offset);
+
+  return kDifOk;
+}
+
+dif_result_t dif_alert_handler_lock_local_alert(
+    const dif_alert_handler_t *alert_handler,
+    dif_alert_handler_local_alert_t local_alert) {
+  if (alert_handler == NULL) {
+    return kDifBadArg;
+  }
+
+#define LOC_ALERT_REGWENS_CASE_(loc_alert_, value_)                       \
+  case loc_alert_:                                                        \
+    regwen_offset = ALERT_HANDLER_LOC_ALERT_REGWEN_##value_##_REG_OFFSET; \
+    break;
+
+  ptrdiff_t regwen_offset;
+  switch (local_alert) {
+    LIST_OF_LOC_ALERTS(LOC_ALERT_REGWENS_CASE_)
+    default:
+      return kDifBadArg;
+  }
+
```

Comment:
nit: `#undef LOC_ALERT_REGWENS_CASE_`

Reply:
And redefine it again under `dif_alert_handler_is_local_alert_locked`. I don't have a strong preference, but if it is a macro that is useful in many functions, then it probably makes sense to move it to the top of this file.

Reply:
Ah, makes sense. Fixed (moved to top of file and #undef'ed at the end).